tiledb/sm/subarray/subarray_partitioner.cc

```diff
@@ -939,9 +939,13 @@ void SubarrayPartitioner::compute_splitting_value_single_range(
   auto cell_order = array_schema->cell_order();
   assert(!range.is_unary());
   auto layout = subarray_.layout();
-  layout = (layout == Layout::UNORDERED || layout == Layout::GLOBAL_ORDER) ?
-               cell_order :
-               layout;
+  if (layout == Layout::UNORDERED && cell_order == Layout::HILBERT) {
```

Comment:
Should this be an `or`? The PR title seems to indicate so?

Reply:
This is a special case for Unordered layout queries with Hilbert cell order arrays. The condition is correct.

Reply:
Thanks for clarify